ouroboros-consensus/src/Ouroboros/Storage/FS/Sim.hs

```diff
@@ -260,14 +281,14 @@ mockSeek (MockHandle fp _mode curOffsetVar) mode offset = withMockFS $ \fs ->
         let offset' = case mode of
              IO.AbsoluteSeek -> offset
              IO.RelativeSeek -> currentOffset + offset
-             IO.SeekFromEnd  -> (toEnum $ BS.length block) - offset
+             IO.SeekFromEnd  -> (toEnum $ BS.length block) + offset
         checkOverflow offset' (BS.length block)
         writeTVar curOffsetVar offset'
         return offset'
   where
     checkOverflow :: DiskOffset -> Int -> ExceptT FsError (Tr m) ()
```

Comment:
Let's just disallow seek from end of the file.

Reply:
@edsko It can have a very specific use case: go manually to the end of file: `fSeek handle SeekFromEnd 0`, that's why I allowed as an option. However I added as a precondition to only SeekFromEnd with offset = 0.